### DISCUSSION TITLE
Exporting function works but No summary of model!!

### DISCUSSION THREAD

**Discussion by Brayan532** [ultralytics/ultralytics#16065] [Q&A]
Hi there,
I have used this model.export(format='saved_model', keras=True) to export the model = YOLO("yolov8n.pt") under Keras. However, after loading by yolo_model = tf.keras.models.load_model('/content/yolov8n_saved_model'), no summary of model can be seen, indeed, an error occurs: AttributeError: '_WrapperFunction' object has no attribute 'summary'. I wonder whether it's the limitation caused by the export function or there is another way to do it? 
I would be happy if someone can give me a hand. 
Best wishes.

**Comment by glenn-jocher** [comment]
@Brayan532 hi there,

The issue you're encountering is likely due to the way the model is wrapped in TensorFlow. To view the summary, try accessing the inner model with `yolo_model.layers[0].summary()`. Let me know if this helps!

**Reply by Brayan532** [reply]
> @Brayan532 hi there,
> 
> The issue you're encountering is likely due to the way the model is wrapped in TensorFlow. To view the summary, try accessing the inner model with `yolo_model.layers[0].summary()`. Let me know if this helps!

Dear Glenn, 
Thanks for your reply. Nonetheless, the same error pops up!
Indeed, my ultimate goal is to integrate a custom model (CNN) to the backbone of yolov8. Since I'm working under Keras, it would be more convenient for me to do that, I reckon. Having said that, I've tried to find a clear example showing the integration of a simple sequential MLP to the backbone of yolo as a reference to me letting me know how to do it. Nevertheless, I've failed to find it. I don't have face to request someone to give me an example, as they might be tied up by their own work. I've also read https://docs.ultralytics.com/usage/cfg/?h=configuration, it's a very good guide, however, I find it confusing at least in the level of my knowledge and the clue I'm looking for. Best wishes.

**Reply by Brayan532** [reply]
Dear @glenn-jocher 
I apologize for any confusion caused by my previous messages regarding the integration of custom models, such as a CNN or modifications to specific layers, within YOLOv8. I would like to kindly request if it would be possible for you or the Ultralytics team to prepare a document or guide that clearly outlines how to add additional layers—such as CNN or MLP layers—into YOLOv8 (e.g., to the backbone) through an example. 

I understand that you and the Ultralytics team are likely very busy, but such a guide would be invaluable to many in the community looking to enhance YOLOv8 in innovative ways. The team has already produced many highly informative and useful documents, and I believe a guide on this topic would be another great contribution.
Thank you for your time and consideration. Warm regards.

**Reply by glenn-jocher** [reply]
Hi @Brayan532,

Integrating a custom model with YOLOv8 in Keras can be complex. While we don't have a specific example for this integration, you might consider using the PyTorch model directly and adapting it to your needs. If you're set on using Keras, you could convert the PyTorch model to ONNX and then to Keras, but this might require additional adjustments. If you have further questions, feel free to ask!

**Reply by Brayan532** [reply]
> Hi @Brayan532,
> 
> Integrating a custom model with YOLOv8 in Keras can be complex. While we don't have a specific example for this integration, you might consider using the PyTorch model directly and adapting it to your needs. If you're set on using Keras, you could convert the PyTorch model to ONNX and then to Keras, but this might require additional adjustments. If you have further questions, feel free to ask!

Hi Glenn, 
Thanks a lot, indeed I meant using the PyTorch directly. Any way,  filled with hope to have Ultralytics supports, I'm gonna initiate the integration directly. Best wishes.

**Reply by glenn-jocher** [reply]
You're welcome! If you encounter any issues during the integration, feel free to reach out for guidance. Best of luck with your project!
